internal/powerplatform/services/environment/api_environment.go

```diff
@@ -435,7 +441,11 @@ func (client *EnvironmentClient) UpdateEnvironment(ctx context.Context, environm
 					return nil, err
 				}
 				tflog.Info(ctx, "Environment State: '"+createdEnv.Properties.States.Management.Id+"'")
-				client.Api.Sleep(3 * time.Second)
+				err = client.Api.SleepWithContext(ctx, 3*time.Second)
```

Comment:
All these hard-coded time durations would probably be better managed in a variable somewhere.